File: backend/app/models/invoice.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import Column, String, Float, Integer, DateTime, Text, Boolean, JSON
from sqlalchemy.orm import relationship
from ..database import Base


class Invoice(Base):
    __tablename__ = "invoices"

    id = Column(String, primary_key=True, default=lambda: f"INV-{uuid.uuid4().hex[:6].upper()}")
    type = Column(String(20), default="client")
    client_name = Column(String(255), nullable=False)
    client_id = Column(String, nullable=True)
    rnc = Column(String(50), default="")
    date = Column(String(20), nullable=False)
    due_date = Column(String(20), default="")
    currency = Column(String(10), default="DOP")
    payment_type = Column(String(20), default="debit")
    payment_method = Column(String(30), default="transfer")
    items = Column(JSON, default=list)
    subtotal = Column(Float, default=0)
    discount = Column(Float, default=0)
    discount_type = Column(String(20), default="percentage")
    discount_amount = Column(Float, default=0)
    taxable_base = Column(Float, default=0)
    tax_rate_id = Column(String(50), default="TAX-001")
    tax = Column(Float, default=0)
    total = Column(Float, default=0)
    status = Column(String(20), default="pending", index=True)
    notes = Column(Text, default="")
    installment_plan = Column(JSON, nullable=True)
    cash_register_id = Column(String, nullable=True)
    amount_received = Column(Float, default=0)
    change_returned = Column(Float, default=0)
    rectifies_id = Column(String, nullable=True)
    created_by = Column(String, nullable=True)
    paid_at = Column(DateTime, nullable=True)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))

File: dotnet-backend/src/Javaline.Commercial.Domain/Entities/Invoice.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Javaline.Commercial.Domain.Common;

namespace Javaline.Commercial.Domain.Entities;

[Table("invoices")]
public class Invoice : BaseEntity
{
    [Key]
    [Column("id")]
    public new string Id { get; set; } = Guid.NewGuid().ToString();

    [Column("type")]
    public string Type { get; set; } = string.Empty;

    [Column("client_name")]
    public string? ClientName { get; set; }

    [Column("client_id")]
    public string? ClientId { get; set; }

    [Column("rnc")]
    public string? Rnc { get; set; }

    [Column("date")]
    public DateTime Date { get; set; } = DateTime.UtcNow;

    [Column("due_date")]
    public DateTime? DueDate { get; set; }

    [Column("currency")]
    public string Currency { get; set; } = "DOP";

    [Column("payment_type")]
    public string? PaymentType { get; set; }

    [Column("payment_method")]
    public string? PaymentMethod { get; set; }

    [Column("items")]
    public string Items { get; set; } = "[]";

    [Column("subtotal")]
    public decimal Subtotal { get; set; }

    [Column("discount")]
    public decimal Discount { get; set; }

    [Column("discount_type")]
    public string? DiscountType { get; set; }

    [Column("discount_amount")]
    public decimal DiscountAmount { get; set; }

    [Column("taxable_base")]
    public decimal TaxableBase { get; set; }

    [Column("tax_rate_id")]
    public string? TaxRateId { get; set; }

    [Column("tax")]
    public decimal Tax { get; set; }

    [Column("total")]
    public decimal Total { get; set; }

    [Column("status")]
    public string Status { get; set; } = "Pending";

    [Column("notes")]
    public string? Notes { get; set; }

    [Column("installment_plan")]
    public string? InstallmentPlan { get; set; }

    [Column("cash_register_id")]
    public string? CashRegisterId { get; set; }

    [Column("amount_received")]
    public decimal? AmountReceived { get; set; }

    [Column("change_returned")]
    public decimal? ChangeReturned { get; set; }

    [Column("rectifies_id")]
    public string? RectifiesId { get; set; }

    [Column("created_by")]
    public string? CreatedBy { get; set; }

    [Column("paid_at")]
    public DateTime? PaidAt { get; set; }
}
